src/Microsoft.DotNet.Interactive.FSharp/FSharpKernel.fs

```diff
@@ -138,6 +140,122 @@ type FSharpKernelBase () as this =
             context.Publish(CompletionsProduced(completionItems, requestCompletions))
         }
 
+    let handleRequestHoverText (requestHoverText: RequestHoverText) (context: KernelInvocationContext) =
+        async {
+            let! (parse, check, _ctx) = script.Value.Fsi.ParseAndCheckInteraction(requestHoverText.Code)
+
+            let res = FsAutoComplete.ParseAndCheckResults(parse, check, EntityCache())
+            let text = FSharp.Compiler.Text.SourceText.ofString requestHoverText.Code
+
+            // seem to be off by one
+            let line = requestHoverText.LinePosition.Line + 1
+            let col = requestHoverText.LinePosition.Character + 1
+
+            let fsiAssemblyRx = System.Text.RegularExpressions.Regex @"^\s*Assembly:\s+FSI-ASSEMBLY\s*$"
+
+            let lineContent = text.GetLineString(line - 1)
+            let! value =
+                async {
+                    match! res.TryGetSymbolUse (FSharp.Compiler.Range.mkPos line col) lineContent with
+                    | Ok (mine, _others) ->
+                        let fullName = 
+                            match mine with
+                            | FsAutoComplete.Patterns.SymbolUse.Val sym ->
+                                match sym.DeclaringEntity with
+                                | Some ent when ent.IsFSharpModule ->   
+                                    match ent.TryFullName with
+                                    | Some _ -> Some sym.FullName
+                                    | None -> None
+                                | _ -> None
+                            | _ ->
+                                None
+
+                        match fullName with
+                        | Some name ->
+                            let expr = if name.StartsWith "Stdin." then name.Substring 6 else name
+                            try return script.Value.Fsi.EvalExpression(expr) |> Some
+                            with _ -> return None
+                        | None -> return None
+
+                    | Error _ ->
+                        return None
+                }
+            
+            match! res.TryGetToolTipEnhanced (FSharp.Compiler.Range.mkPos line col) lineContent with
```

Comment:
Yes I'm wondering why we need FsAutoComplete tooltips here, why not the tooltips provided by FCS.  That might reduce the amount of code copied across (and leave us to incorporate the FsAutoComplete tooling into FCS).

Reply:
I first tried to use the plain tooltips from FCS but they don't incorporate types/modules etc. (at least when used with overlapping names).
When hovering the type `int` i got the tooltip for the `int : ^T -> int` function, so i figured FsAutoComplete could be a way to go.

Reply:
> I first tried to use the plain tooltips from FCS but they don't incorporate types/modules etc. (at least when used with overlapping names).

You're right that the FsAutoComplete tooltips make improvements (though it's not yet clear to me if they also regress other things).

But in any case, for .NET Interactive we would take the approach that we build on FSharp.Compiler.Service and just have to accept the tooling possibilities that gives.  If we need to improve those tooltips, let's do it in FCS (e.g. by incorporating and aligning the tooltips code with Ionide)